tests/include/ApiTest.php

```diff
@@ -44,24 +44,45 @@ class ApiTest extends DatabaseTest
 	/** @var App */
 	protected $app;
 
+	/** @var Configuration */
+	protected $config;
+
+	/** @var Dice */
+	protected $dice;
+
 	/**
 	 * Create variables used by tests.
 	 */
 	protected function setUp()
 	{
 		parent::setUp();
 
-		$dice = new Dice();
-		$dice = $dice->addRules(include __DIR__ . '/../../static/dependencies.config.php');
-		$dice = $dice->addRule(Database::class, ['instanceOf' => StaticDatabase::class, 'shared' => true]);
-		BaseObject::setDependencyInjection($dice);
+		$this->dice = new Dice();
+		$this->dice = $this->dice->addRules(include __DIR__ . '/../../static/dependencies.config.php');
```

Comment:
Do you have to assign the class member three times? Can't you do something like:
```php
$this->dice = (new Dice())
    ->addRules(include __DIR__ . '/../../static/dependencies.config.php')
    ->addRule(Database::class, ['instanceOf' => StaticDatabase::class, 'shared' => true]);
```
Or at least 
```php
$this->dice = new Dice();
$this->dice->addRules(include __DIR__ . '/../../static/dependencies.config.php');
$this->dice->addRule(Database::class, ['instanceOf' => StaticDatabase::class, 'shared' => true]);
```
?

Reply:
Dice is immutable, means that every new rule adds a new instance of the class.
https://github.com/Level-2/Dice/blob/master/README.md#1-dice-is-immutable
But I think the first command should work